# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/Array.cpp
# Commit: 9bfa6023a553
# Full Hash: 9bfa6023a5539598e824e3d2a95dec25d099a378
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 5 - Remove JSClass* from ObjectGroup. r=jonco
#   
#   BaseShape stores the JSClass* too so this patch changes the code to get the JSClass*
#   from there instead of from the ObjectGroup.
#   
# ==============================================================================

diff -r 2186bd51bebd -r 9bfa6023a553 js/src/builtin/Array.cpp
--- a/js/src/builtin/Array.cpp	Sat Mar 06 01:05:20 2021 +0000
+++ b/js/src/builtin/Array.cpp	Sat Mar 06 01:05:20 2021 +0000
@@ -3980,9 +3980,10 @@
   }
 
   AutoSetNewObjectMetadata metadata(cx);
-  RootedArrayObject arr(cx, ArrayObject::createArray(
-                                cx, allocKind, GetInitialHeap(newKind, group),
-                                shape, group, length, metadata));
+  RootedArrayObject arr(
+      cx, ArrayObject::createArray(
+              cx, allocKind, GetInitialHeap(newKind, &ArrayObject::class_),
+              shape, group, length, metadata));
   if (!arr) {
     return nullptr;
   }
@@ -4070,7 +4071,7 @@
   RootedObjectGroup group(cx, templateObject->group());
   RootedShape shape(cx, templateObject->lastProperty());
 
-  gc::InitialHeap heap = GetInitialHeap(GenericObject, group);
+  gc::InitialHeap heap = GetInitialHeap(GenericObject, &ArrayObject::class_);
   Rooted<ArrayObject*> arr(
       cx, ArrayObject::createArray(cx, allocKind, heap, shape, group, length,
                                    metadata));